src/Templates/SingleProductTemplateCompatibility.php

```diff
@@ -241,10 +241,11 @@ protected function set_hook_data() {
 			),
 			'woocommerce_after_single_product_summary'  => array(
 				'block_name' => 'woocommerce/product-details',
-				'position'   => 'before',
+				'position'   => 'after',
 				'hooked'     => array(
 					'woocommerce_output_product_data_tabs' => 10,
-					'woocommerce_upsell_display'           => 15,
+					// We want to display the upsell products after the last block that belongs to the Single Product.
+					// 'woocommerce_upsell_display'           => 15.
```

Comment:
This tested nicely however it seems very counterintuitive that commenting out the `woocommerce_upsell_display` is the solution to display `upsells`. Can you explain this?

Reply:
> This PR restores the action that renders the grid that includes the upsell products.

The Compatibility Layer removes the default callbacks added to hooks by WooCommerce. This is necessary to avoid duplicate content (since that with the blockified templates, blocks render the HTML). As discussed in #9610, it could not sense that the merchant should see/add a block for product customization on the Single Product Template (a generic template). Since that, we want to enable the blockified templates in the next release, I think that enabling the default callbacks that render the upsells, is a good approach, at least for now.